### PR TITLE
fix(deps): Update react date picker and adjust usage

### PR DESCRIPTION
**Pull request by darryltec**
## Motivations

Updating react-datepicker to latest throws an error on one of the props we're using.

image
[JOB-61503](https://jobber.atlassian.net/browse/JOB-61503)

This aims to fix that issue by

### Added

- 

### Changed

- Replacing `formatWeekDay` call back to get the first 3 letters of the week to `useWeekdaysShort`. 
  - Which is a built-in feature of the package to do exactly what we want.
- Dute to some unforeseen 3rd party change, we need to declare the date formats to allow users to type month words

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

[JOB-61503]: https://jobber.atlassian.net/browse/JOB-61503?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by darryltec** [comment]
Closing this as the 3rd party package seems to have removed a feature we're using

**Review by danlaurent — APPROVED** [review]
Tested and it worked as expected! LGTM! 🚀

File: packages/components/src/InputDate/InputDate.test.tsx [hunk 1]
```diff
@@ -85,22 +85,6 @@ it("updates the value of the field when the value prop changes", () => {
...
-  );
-  expect(getByDisplayValue(date)).toBeInTheDocument();
-
-  fireEvent.change(getByDisplayValue(date), {
-    target: { value: newDate },
-  });
-
-  expect(changeHandler).toHaveBeenCalledWith(new Date(newDate));
-});
-
```
[the remark this answers is not in the document]
Comment (darryltec) [reply packages/components/src/InputDate/InputDate.test.tsx]: We back in bizniz

File: packages/components/src/DatePicker/DatePicker.tsx [hunk 2]
```diff
@@ -115,13 +115,14 @@ export function DatePicker({
...
         minDate={minDate}
-        formatWeekDay={date => date.substring(0, 3)}
+        useWeekdaysShort={true}
         customInput={
           <DatePickerActivator activator={activator} fullWidth={fullWidth} />
         }
         renderCustomHeader={props => <DatePickerCustomHeader {...props} />}
         onCalendarOpen={handleCalendarOpen}
         onCalendarClose={handleCalendarClose}
+        dateFormat={["P", "PP", "PPP"]}
```
Comment (darryltec) [inline packages/components/src/DatePicker/DatePicker.tsx:125]: Allow system date format (P), short months (PP, `Feb`), long months (PPP, `February`)

**Comment by darryltec** [comment]
sorry @danlaurent let's do it one more time haha

**Comment by chris-at-jobber** [comment]
Is there a way to bring back the non-requirement of commas?

On the left (production) you can see I've entered `nov 21 2011` and it's selected the correct date.

On the right (this branch) I've entered the same value and it's not selected a date

image

Once I add the comma to read `nov 21, 2011` it works, but this doesn't seem intuitive

image

**Comment by darryltec** [comment]
@chris-at-jobber totes can, try now (or when it deploys)

**Review by danlaurent — APPROVED** [review]
Tested the new changes, lgtm!

**Review by chris-at-jobber — COMMENTED** [review]
Design approved - tested for parity with current component, thanks for flagging this and making sure we didn't lose any functionality!

image

**Comment by darryltec** [comment]
Noice. I probably should do a pre-release for this and test it against Jobber. Hopefully there aren't any surprises 🤞🏼

**Comment by darryltec** [comment]
ayt we good!

**Pull request merged**